fragmenter/fragment.py

```diff
@@ -1,818 +1,652 @@
-from itertools import combinations
-import cmiles.utils
-from cmiles.utils import mol_to_smiles, has_stereo_defined, has_atom_map, is_missing_atom_map, remove_atom_map, restore_atom_map
+import abc
+import logging
+import time
+from collections import defaultdict
+from typing import Any, Dict, List, Literal, Optional, Set, Tuple, Union
```

Comment:
Also, just in case you plan to support Python < 3.8, I guess you may want to move `Literal` to import from typing-extensions.

Reply:
Ah good catch - thanks!

Reply:
Resolved in 014af99